### PR TITLE
[radiothermostat] Ignore updates if thermostat data is invalid

### PR DESCRIPTION
**Pull request by mlobstein**
It could be that my thermostat is beginning to fail since it has started randomly returning -1 instead of the actual temperature. This change will cause a polling update with invalid data to be ignored so it will not update the channels. As a result, the temperature graphs will no longer look like this:

temp dropout

### CODE REVIEW

File: bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java [hunk 1]
```diff
@@ -371,14 +371,18 @@ public void onNewMessageEvent(RadioThermostatEvent event) {
             switch (evtKey) {
                 case DEFAULT_RESOURCE:
                     rthermData.setThermostatData(gson.fromJson(evtVal, RadioThermostatTstatDTO.class));
-                    updateAllChannels();
+                    // if thermostat returned -1 for temperature, skip this update
+                    if (rthermData.getThermostatData().getTemperature() > 0) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java:375]: Maybe rather >= 0

File: bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java [hunk 2]
```diff
@@ -371,14 +371,18 @@ public void onNewMessageEvent(RadioThermostatEvent event) {
...
+                        updateAllChannels();
+                    }
                     break;
                 case HUMIDITY_RESOURCE:
                     RadioThermostatHumidityDTO dto = gson.fromJson(evtVal, RadioThermostatHumidityDTO.class);
-                    if (dto != null) {
+                    // if thermostat returned -1 for humidity, skip this update
+                    if (dto != null && dto.getHumidity() > 0) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java:382]: Maybe rather >= 0

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java (hunk 1 above)
Comment (mlobstein) [reply bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java]: ok, fixed

File: bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java (hunk 2 above)
Comment (mlobstein) [reply bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java]: fixed

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**